src/include/function/table/bind_input.h

```diff
@@ -12,20 +12,24 @@ class ClientContext;
 
 namespace function {
 
+struct TableFunction;
 struct ScanTableFuncBindInput {
     std::vector<common::Value> inputs;
     common::ReaderConfig config;
     std::vector<std::string> expectedColumnNames;
     std::vector<common::LogicalType> expectedColumnTypes;
     main::ClientContext* context;
+    function::TableFunction* tableFunction;
```

Comment:
This is a bit weird in the sense that `tableFunction` is not required as a bindInput. Instead, as what I see in this PR, u are trying to set `parallelFunc` inside `bindFunc` and `parallelFunc` is only available inside `tableFunction`.

I would try a different approach where we don't change `ScanTableFuncBindInput`. Instead, we add canParallel as field inside `TableFuncBindData` and config `tableFunction` after binding.